### DISCUSSION TITLE
Orchard CMS page's html source > 50MB???

### DISCUSSION THREAD

**Discussion by Canoff13** [OrchardCMS/OrchardCore#8032] [General]
(I got further dev tasks on legacy site).

While other Orchard sites on the Net, have less then 1MB html source data per page with similar complexity, this Orchard site pages are over 50MB!

Pages are loading damned slowly. Looking at the html source of these very simple pages, it turned out that they are more than 50MB (/page), and are full of `shapeTracingMetadataHost `objects having a lots of (needless?) data in the `placement`, `html`, and `model `properties.

Part of this stuff is rendered into the model is in the `Orchard.Designer.Tools\Services\TemplatesFilter.cs`, in this lines:
`var tail = _workContext.Layout.Tail; 
tail.Add(_shapeFactory.ShapeTracingTemplates());`

One of the place where this data is written into the page in `Views\Document.cshtml` from `Model.Tail.Items' Dump` property.

_What is it?
What is the purpose of it?
Can it be switched off?_

@Piedone Any idea?

Another interesting thing, that the page looks seemingly downloaded, but the browser's still showing that it is not ready for at least one more minute and cause high CPU load too.

Reloading the page often cause Out of memory error in `Document.cshtml `of the customized Theme at this line of code:
`@Display(Model.Tail)`

Appreciate your help

**Comment by deanmarcussen** [comment]
For Orchard 1 Legacy sites, it is better to open an issue in the Orchard 1 repository here https://github.com/OrchardCMS/Orchard

However your problem is that someone has turned on the Shape Tracing feature and left it on. You need to turn the feature off.

**Reply by Canoff13** [reply]
This site uses Orchard v.1.10.1.0
I meant 'legacy' as developed by former employees and therefore no support info available : )

Thank you for the information. I guessed a similar thing in the background.

Do I need the Designer Tools to switch it off?

**Reply by Piedone** [reply]
The feature under Modules/Features is literally called "Shape Tracing":
![image](https://user-images.githubusercontent.com/1976647/102386777-d89ed480-3fcf-11eb-9e63-5d08edc24fdb.png)
This is what you have to disable.

**Reply by Canoff13** [reply]
What a difference! : -)

66 568KB -> 95KB

Thank you all
